src/main/scala/firrtl/analyses/CircuitGraph.scala

```diff
@@ -0,0 +1,137 @@
+// See LICENSE for license details.
+
+package firrtl.analyses
+
+import firrtl.Kind
+import firrtl.annotations.TargetToken.{Instance, OfModule}
+import firrtl.annotations._
+import firrtl.ir.{Circuit, DefInstance}
+
+/** Use to construct [[CircuitGraph]]
+  * Also contains useful related functions
+  */
+object CircuitGraph {
+
+  /** Build a CircuitGraph
+    * [[firrtl.ir.Circuit]] must be of MiddleForm or lower
+    * @param circuit
+    * @return
+    */
+  def apply(circuit: Circuit): CircuitGraph = new CircuitGraph(ConnectionGraph(circuit))
+
+  /** Return a nicely-formatted string of a path of [[firrtl.annotations.ReferenceTarget]]
+    * @param path
+    * @param tab
+    * @return
+    */
+  def prettyToString(path: Seq[ReferenceTarget], tab: String = ""): String = {
```

Comment:
I wonder if this `path` is confusing, because we use the same word to mean either the instance path to a non-local reference, or a connection path through a graph. Perhaps change it to `connectionPath` or something like that?